llvm/lib/ProfileData/Coverage/CoverageMapping.cpp

```diff
@@ -436,8 +439,9 @@ Expected<std::unique_ptr<CoverageMapping>> CoverageMapping::load(
       if (PathOpt) {
```

Comment:
`BIDFetcher->fetch(BinaryID);` should return the unstripped file with debug info using either local filesystem or debuginfod, see

https://github.com/llvm/llvm-project/blob/265d48a02d2453aaf2d29eefbfedd3595220f4e6/llvm/lib/Object/BuildID.cpp#L68
https://github.com/llvm/llvm-project/blob/dc925be68b2fb214a30cce86b312f469bdb313b0/llvm/lib/Debuginfod/BuildIDFetcher.cpp#L22

If `PathOpt` contains value and `DebugInfoFilename` is an empty string, we should be using `*PathOpt` as the debug info filename.